apps/src/lib/node/ledger/shell/prepare_proposal.rs

```diff
@@ -185,16 +179,23 @@ mod prepare_block {
             let mut signatures = HashMap::new();
 
             let total_voting_power =
-                self.get_total_voting_power(Some(events_epoch)).into();
-            let mut voting_power = 0u64;
+                u64::from(self.get_total_voting_power(Some(events_epoch)));
+            let mut voting_power = FractionalVotingPower::default();
+
+            let deserialized = deserialize_vote_extensions(vote_extensions);
 
             for (validator_voting_power, vote_extension) in
-                self.filter_invalid_vote_extensions(vote_extensions)
+                self.filter_invalid_vote_extensions(deserialized)
             {
                 let validator_addr = vote_extension.data.validator_addr;
 
                 // update voting power
-                voting_power += u64::from(validator_voting_power);
+                let validator_voting_power = u64::from(validator_voting_power);
+                voting_power += FractionalVotingPower::new(
+                    validator_voting_power,
+                    total_voting_power,
+                )
+                .unwrap_or_default();
```

Comment:
Yeah, something has gone terribly wrong if our storage was corrupted in such a way that creating a new `FractionalVotingPower` would fail. I'll add an expect here.